Creates a README

kikosgc opened the pull request:
Creates a README for SAE J1979 OBDb Repo

@jverkoey

kikosgc committed:
Creates a README

Creates a README for SAE J1979 OBDb Repo

jverkoey commented on README.md:
```diff
@@ -1,2 +1,42 @@
+- Developers creating diagnostic tools and software for reading OBD-II data.
+- Enthusiasts interested in understanding vehicle performance data.
+
+## SAE J1979 PIDs
```
Suggested modification:

```

## SAE J1979 PIDs and Commands

The SAE J1979 standard includes a range of Parameter IDs (PIDs) that are used to request specific data from a vehicle's Engine Control Module (ECM). In this repository, we organize these PIDs as **Commands** - structured requests that include both the PID and additional metadata needed to properly communicate with the vehicle.

### Commands and Signals

Each Command represents a complete OBD-II message that can be sent to the vehicle and includes:

- **PID Code**: The specific Parameter ID being requested
- **Header (HDR)**: The command's 11-bit identifier for CAN communication
- **Frequency**: How often the command can be safely sent to the vehicle
- **Signals**: The data points that the vehicle returns in response to this command

When a Command is sent to the vehicle, it responds with one or more **Signals**. While PIDs are often thought of as representing specific data points, it's more accurate to think of them as requests that can return multiple pieces of information (Signals). For example, a single PID command might return multiple Signals such as:

- Engine RPM
- Throttle position
- Engine load

Each Signal includes:

- **Identifier**: A unique name for the data point
- **Format**: Instructions for interpreting the raw bytes from the vehicle's response
- **Units**: The measurement units for the decoded value
- **Description**: What the data represents

### Example Structure

```json
{
  "commands": [{
    "hdr": "7E0",
    "cmd": {"01": "0C"},
    "freq": 0.1,
    "signals": [{
      "id": "ENGINE_RPM",
      "fmt": {
        "len": 16,
        "max": 16383.75,
        "mul": 0.25,
        "unit": "revolutionsPerMinute"
      },
      "name": "Engine RPM"
    }]
  }]
}
```

kikosgc replied:
Updated!

jverkoey reviewed (changes requested):
Thank you! This is a good start.

kikosgc committed:
Updates README

- Fix typo
- Re-works the "SAE J1979 PIDs" section with recommendations of @jverkoey

Co-Authored-By: Jeff Verkoeyen <jeff@clutch.engineering>

jverkoey committed:
Update README.md
